cmd/buildah/bud.go

```diff
@@ -115,14 +115,10 @@ func budCmd(c *cli.Context) error {
 	if len(dockerfiles) == 0 {
 		dockerfiles = append(dockerfiles, filepath.Join(contextDir, "Dockerfile"))
 	}
-	if err := parse.ValidateFlags(c, buildahcli.BudFlags); err != nil {
-		return err
-	}
-	if err := parse.ValidateFlags(c, buildahcli.LayerFlags); err != nil {
-		return err
-	}
-	if err := parse.ValidateFlags(c, buildahcli.FromAndBudFlags); err != nil {
-		return err
+	for _, flag := range [][]cli.Flag{buildahcli.BudFlags, buildahcli.LayerFlags, buildahcli.FromAndBudFlags} {
+		if err := parse.ValidateFlags(c, flag); err != nil {
+			return err
```

Comment:
@boaz1337 my only concern is with the old code, if we had an error it would say "Error line 122: {err}" or similar and we'd know that the failure happened when checking the LayerFlags.  With this change would we know which had an issue (BudFlags, LayerFlags or FromAndBudFlags)?  It shouldn't be a problem in production, most likely something we'd hit during development/testing.

Reply:
Yeah, I see what you're saying.
Maybe adding index to the error, will help?
I don't mind closing this pull request if you want to.

Reply:
The thing is I really like the change from a code efficiency standpoint.  Anyway to have the flag type that was being checked in the error....